### PR TITLE
API-2934: faking XML as HAL

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala [hunk 1]
```diff
@@ -26,9 +26,9 @@ import play.api.test.Helpers._
 
 class RetrieveAndDeleteNotificationSpec extends FeatureSpec with GivenWhenThen with Matchers with GuiceOneAppPerTest {
 
-  feature("Retrieve a single message from the Pull Service (DELETE)") {
+  feature("Retrieve(DELETE) a single message from the API Notification Pull service") {
     info("As a 3rd Party")
-    info("I want to successfully receive any notifications waiting for me")
+    info("I want to successfully receive all notifications waiting for me")
```
Comment (stewabel) [inline test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala]: This test covers retrieving a single notification. Perhaps this should read something like `info("I want to successfully receive a notification waiting for me")`

Comment (gokyo) [reply test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala]: fair enough

File: test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala [hunk 2]
```diff
@@ -26,9 +26,9 @@ import play.api.test.Helpers._
 
 class RetrieveAndDeleteNotificationSpec extends FeatureSpec with GivenWhenThen with Matchers with GuiceOneAppPerTest {
 
-  feature("Retrieve a single message from the Pull Service (DELETE)") {
+  feature("Retrieve(DELETE) a single message from the API Notification Pull service") {
```
Comment (stewabel) [inline test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala]: Would the cleanup of this file be better on a separate commit on the `API-2932` branch? These tests relate to that ticket, and could cause a merge conflict, as there are a lot more changes to this spec on that branch

File: app/uk/gov/hmrc/apinotificationpull/util/XmlBuilder.scala [hunk 3]
```diff
@@ -19,14 +19,6 @@ package uk.gov.hmrc.apinotificationpull.util
 import uk.gov.hmrc.apinotificationpull.model.Notifications
 
 object XmlBuilder {
```
Comment (stewabel) [inline app/uk/gov/hmrc/apinotificationpull/util/XmlBuilder.scala]: In API-2932, I created a package named `presenters` which contains a `NotificationPresenter` Its job is very similar to this builder. Perhaps this could be called `NotificationsXmlPresenter` and live in the same package?

Comment (gokyo) [reply app/uk/gov/hmrc/apinotificationpull/util/XmlBuilder.scala]: I would use a `util` package rather than `presenter`.
Let's see what other people think

File: test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala (hunk 2 above)
Comment (gokyo) [reply test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala]: Fair enough
I will move this to your branch
